vsphere_plugin_common/__init__.py

```diff
@@ -274,13 +280,197 @@ def _get_port_group_names(self):
+                    )
+                )
+
+        if allowed_datastores:
+            ctx.logger.debug('Checking allowed datastores list.')
+            datastore_list = self.get_datastores()
+            datastore_names = [datastore.name for datastore in datastore_list]
+
+            not_datastores = set(allowed_datastores).difference(
+                set(datastore_names)
+            )
+            if len(not_datastores) == len(allowed_datastores):
+                issues.append(
+                    'No allowed datastores exist. Allowed datastore(s): '
+                    '{allow}. Existing datastore(s): {datastores}.'.format(
+                        allow=', '.join(allowed_datastores),
+                        datastores=', '.join(datastore_names),
+                    )
+                )
+            elif len(not_datastores) > 0:
+                ctx.logger.warn(
+                    'One or more specified allowed datastores do not exist: '
+                    '{not_datastores}.'.format(
+                        not_datastores=', '.join(not_datastores),
+                    )
+                )
+
+        ctx.logger.debug('Checking template exists.')
+        template_vm = self._get_obj_by_name([vim.VirtualMachine],
+                                            template_name)
+        if template_vm is None:
+            issues.append("VM template {0} could not be found.".format(
+                template_name
+            ))
+
+        ctx.logger.debug('Checking resource pool exists.')
+        resource_pool = self._get_obj_by_name(
+            [vim.ResourcePool],
+            resource_pool_name,
+        )
+        if resource_pool is None:
+            issues.append("Resource pool {0} could not be found.".format(
+                resource_pool_name,
+            ))
+
+        ctx.logger.debug('Checking datacenter exists.')
+        datacenter = self._get_obj_by_name([vim.Datacenter],
+                                           datacenter_name)
+        if datacenter is None:
+            issues.append("Datacenter {0} could not be found.".format(
+                datacenter_name
+            ))
+
+        ctx.logger.debug(
+            'Checking networks exist.'
+        )
+        port_groups, distributed_port_groups = self._get_port_group_names()
+        for network in networks:
+            network_name = network['name']
+            network_name_lower = network_name.lower()
+            switch_distributed = network['switch_distributed']
+
+            list_distributed_networks = False
+            list_networks = False
+            # Check network exists and provide helpful message if it doesn't
+            # Note that we special-case alerting if switch_distributed appears
+            # to be set incorrectly.
+            # Use lowercase name for comparison as vSphere appears to be case
+            # insensitive for this.
+            if switch_distributed:
+                error_message = (
```

Comment:
Otherwise it goes over 80 characters. Seemed cleaner to break it like this rather than breaking it mid-sentence.